### PR TITLE
notary: init at 0.5.1

### PR DESCRIPTION
**Pull request by vdemeester**
Signed-off-by: Vincent Demeester <vincent@sbr.pm>

###### Motivation for this change

 Notary is a project that allows anyone to have trust over arbitrary collections of data. It is used for docker content trust but definitely can be used on its own.

https://github.com/theupdateframework/notary

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/tools/security/notary/default.nix [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+              sha256 = "1e32aad049d2835b0c5b4805f9abc39b88c4de90b9743775c33c9faaf6c6c383"; };
+in stdenv.mkDerivation rec {
+    shortname = "notary";
+    name = "${shortname}-${version}";
+    version = "0.4.3";
+
+    src = fetchurl {
+        url = "https://github.com/theupdateframework/notary/releases/download/v${version}/${shortname}-${metadata.arch}";
+        sha256 = metadata.sha256;
+    };
```
Comment (Mic92) [inline pkgs/tools/security/notary/default.nix]: Any reason not to compile from source? The build looks straight forward:
https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=notary-git

Comment (vdemeester) [reply pkgs/tools/security/notary/default.nix]: That was to go as quick as I needed.. I can take a look at compiling for source :wink:

File: pkgs/tools/security/notary/default.nix [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        longDescription = ''
+          The Notary project comprises a server and a client for running and interacting with trusted collections. See the service architecture documentation for more information.
+
+          Notary aims to make the internet more secure by making it easy for people to publish and verify content. We often rely on TLS to secure our communications with a web server which is inherently flawed, as any compromise of the server enables malicious content to be substituted for the legitimate content.
+
+          With Notary, publishers can sign their content offline using keys kept highly secure. Once the publisher is ready to make the content available, they can push their signed trusted collection to a Notary Server.
+
+          Consumers, having acquired the publisher's public key through a secure channel, can then communicate with any notary server or (insecure) mirror, relying only on the publisher's key to determine the validity and integrity of the received content.
+        '';
+        license = licenses.apache2;
```
Comment (grahamc) [inline pkgs/tools/security/notary/default.nix]: I think you want `asl2` instead of `apache2`.

Comment (vdemeester) [reply pkgs/tools/security/notary/default.nix]: oups 😅

**Comment by vdemeester** [comment]
Updated (building from sources, rebased)

File: pkgs/tools/security/notary/default.nix [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  src = fetchgit {
+    url = "https://github.com/theupdateframework/notary";
+    rev = "refs/tags/v${version}";
+    sha256 = "1max6rgangmw892l8zpijhx5bj9zllqim457k1dq7vf1mzybihr6";
+    leaveDotGit = true;
+  };
+
+  buildInputs = [ git libtool ];
+
+  goPackagePath = "github.com/docker/notary";
```
Comment (vdemeester) [inline pkgs/tools/security/notary/default.nix]: The move to `theupdateframework` (due to giving the project to CNCF) happened after the `v0.4.3` release.. so building imports, etc.. are still refering the path with `docker`.

File: pkgs/tools/security/notary/default.nix [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+buildGoPackage rec {
+  name = "notary-${version}";
+  version = "0.4.3";
+  gitcommit = "9211198";
+
+  src = fetchgit {
+    url = "https://github.com/theupdateframework/notary";
+    rev = "refs/tags/v${version}";
+    sha256 = "1max6rgangmw892l8zpijhx5bj9zllqim457k1dq7vf1mzybihr6";
+    leaveDotGit = true;
```
Comment (Mic92) [inline pkgs/tools/security/notary/default.nix]:
```
output path '/nix/store/g0x3f272dhwns82v2ygaa3w72y2gbhkp-notary' has r:sha256 hash '1j75973a6vciwcrqxbzvk6smr5s502dsa43723g3k9f33nz8lyhz' when '1max6rgangmw892l8zpijhx5bj9zllqim457k1dq7vf1mzybihr6' was expected
```
Looks like the checksum of `fetchgit` changed or my systems creates a different one then yours.
Can we get rid of `leaveDotGit`?

Comment (Mic92) [reply pkgs/tools/security/notary/default.nix]: It should be possible to set `make GITCOMMIT=<hash> GITUNTRACKEDCHANGES= ` and get rid of the git repository:
https://github.com/theupdateframework/notary/blob/master/Makefile#L8

File: pkgs/tools/security/notary/default.nix [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+  };
+
+  buildInputs = [ git libtool ];
+
+  goPackagePath = "github.com/docker/notary";
+  buildPhase = ''
+    cd go/src/github.com/docker/notary
+    make GITCOMMIT=${gitcommit} client
```
Comment (Mic92) [inline pkgs/tools/security/notary/default.nix]: ah, you are already doing that. so only `GITUNTRACKEDCHANGES=` is missing.

**Comment by Mic92** [comment]
@GrahamcOfBorg build notary

**Comment by vdemeester** [comment]
huh ? the tag changed 😱

**Comment by vdemeester** [comment]
Updated to `0.5.1` 🤔

**Pull request merged**